core/src/main/java/org/elasticsearch/index/shard/TranslogRecoveryPerformer.java

```diff
@@ -152,6 +155,7 @@ public void performRecoveryOperation(Engine engine, Translog.Operation operation
                                     .routing(create.routing()).parent(create.parent()).timestamp(create.timestamp()).ttl(create.ttl()),
                             create.version(), create.versionType().versionTypeForReplicationAndRecovery(), Engine.Operation.Origin.RECOVERY);
                     maybeAddMappingUpdate(engineCreate.type(), engineCreate.parsedDoc().dynamicMappingsUpdate(), engineCreate.id(), allowMappingUpdates);
+                    indexingService.preCreate(engineCreate);
```

Comment:
ok I'll need to investigate a bit more to understand the potential consequences of this change :)